Sphdem

dineshadepu opened the pull request.

prabhuramachandran commented on pysph/examples/rigid_body/rigid_block_in_tank.py:
```diff
@@ -1,191 +1,250 @@
-# NumPy
+from __future__ import print_function
 import numpy as np
+import matplotlib.pyplot as plt
```
The test builders may not have matplotlib so it is better to have this import inside the post_process method.

prabhuramachandran commented on pysph/sph/rigid_body.py:
```diff
@@ -299,49 +300,271 @@ def loop(self, d_idx, d_m, d_rho, d_au, d_av, d_aw,  d_p,
+
+class LiuFluidForce(Equation):
+    """Force between a solid sphere and a SPH fluid particle.
+    This is implemented using Akinchi's
```
Spelling of Akinchi should be Akinci.

dineshadepu replied:
Because I added additional force due to solid particle, which is not given in Akinci's paper. 

Now I added additional documentation about that.

prabhuramachandran commented on pysph/examples/rigid_body/rigid_block_in_tank.py:
```diff
@@ -1,191 +1,250 @@
-"""A block of rigid solid falling on a tank of water. (6 minutes)
+"""A block of rigid solid falling on wood in a tank of water.
```
Can you please retain and fix the original simpler example and add your new one as a separate example?  Thanks.

prabhuramachandran commented on pysph/sph/rigid_body.py:
```diff
@@ -299,49 +300,271 @@ def loop(self, d_idx, d_m, d_rho, d_au, d_av, d_aw,  d_p,
         s_fz[s_idx] += -d_m[d_idx]*az
 
+
+class LiuFluidForce(Equation):
```
Why is this called LiuFluidForce?

dineshadepu replied:
Because I added additional force due to solid particle, which is not given in Akinci's paper.

Now I added additional documentation about that

prabhuramachandran commented on pysph/sph/rigid_body.py:
```diff
@@ -299,49 +300,271 @@ def loop(self, d_idx, d_m, d_rho, d_au, d_av, d_aw,  d_p,
-    """This is inspired from
-    http://http.developer.nvidia.com/GPUGems3/gpugems3_ch29.html
-    and
+    """Force between two spheres is implemented using DEM contact force law.
```
Can you again keep the original force implemented in another equation?  Perhaps `RigidBodyForceGPUGems` or something appropriate?  And add yours as the `RigidBodyCollision` one?

prabhuramachandran commented on pysph/examples/rigid_body/rigid_block_in_tank.py:
```diff
@@ -1,191 +1,250 @@
+    plt.show()
+
+
+class FluidStructureInteration(Application):
```
Well this is really not FSI but more rigid fluid interaction.

prabhuramachandran commented on pysph/examples/rigid_body/cubes_colliding_in_tank.py:
```diff
@@ -0,0 +1,261 @@
+from pysph.solver.solver import Solver
+from pysph.sph.integrator import EPECIntegrator
+from pysph.sph.integrator_step import WCSPHStep
+from pysph.solver.utils import get_files, iter_output
```
These don't seem used.  Can you please remove unused imports?

prabhuramachandran commented on pysph/examples/rigid_body/solid_body_floating_in_tank.py:
```diff
@@ -0,0 +1,217 @@
+from pysph.solver.solver import Solver
+from pysph.sph.integrator import EPECIntegrator
+from pysph.sph.integrator_step import WCSPHStep
+from pysph.solver.utils import get_files, iter_output
```
Again, unused imports.

prabhuramachandran commented on pysph/examples/rigid_body/solid_body_floating_in_tank.py:
```diff
@@ -0,0 +1,217 @@
+                                 alpha=self.alpha, beta=0.0, c0=self.co,
+                                 gy=-9.81),
+                LiuFluidForce(dest='fluid', sources=['cube'],),
+                # PressureRigidBody(dest='fluid', sources=['cube'],
```
Why do you have commented code here?  Please remove.

prabhuramachandran commented on pysph/examples/rigid_body/rigid_block_in_tank.py:
```diff
@@ -133,55 +140,60 @@ def create_particles(self):
+                                 sources=['fluid', 'solid', 'block'],
+                                 alpha=alpha, beta=beta, gy=-9.81, c0=c0,
+                                 tensile_correction=True),
+                PressureRigidBody(dest='fluid', sources=['block', 'solid'],
```
Should this use LiuFluidForce instead?  Please make any changes to this example to make it work correctly.  Thanks.

dineshadepu replied:
Please ignore this example.

prabhuramachandran replied:
Well if the example is not useful and produces bad results do we really need it?

dineshadepu replied:
I think we can remove that example. Even the implementation is a little complex to fix it.

prabhuramachandran commented on pysph/examples/rigid_body/solid_body_floating_in_tank.py:
```diff
@@ -0,0 +1,217 @@
+    plt.show()
+
+
+class RigidFluidCOupling(Application):
```
Spelling of class "COupling"

prabhuramachandran commented on pysph/examples/rigid_body/solid_body_floating_in_tank.py:
```diff
@@ -0,0 +1,217 @@
+"""A sphere of density 500 falling into a hydrostatic tank (15 minutes)
+
+Check basic equations of SPH to throw a ball inside the vessel
```
Is this different from rigid_block_in_tank.py?  If not just retain any one of the correct examples.

dineshadepu replied:
Yes. Here the density is 500.  It is implemented using new formulation.

dineshadepu commented:
Dear sir,
I made changes as per your request. I made the following changes:

- I added the liu fluid force documentation
- I retained the GPUGems force.
- I added two more examples on Rigid Fluid coupling. 
    - 500 density cube floating
    - Two cubes hitting in a tank
- I haven't fixed the original rigid block in tank. Even it is retained.

Thanks.

dineshadepu commented:
Please ignore rigid_body_in_tank.py example. I haven't fixed it.

prabhuramachandran commented:
But isn't the example very similar to the other one?  Should it not work if you replace the force with the new LiuFluidForce?

dineshadepu commented:
Not only force equations, I have to add many other particle properties to the particle arrays. 
Give me some time I will do it.

Thanks.

prabhuramachandran commented:
@dineshadepu -- no need if you think it is pointless and the other one works well.  I was just checking, if you feel that the new example is fine and works well, go ahead and remove the old one.

prabhuramachandran commented:
Thanks for all the changes!
